### PR TITLE
Feature video playback

### PR DESCRIPTION
**Pull request by Cruzo007**
- Enabling the Video Playback feature.
- Added README/Repo links to popup.
- Added default playback asset around 300kb in size.
- Incrementing version to 0.2 series, given the new feature addition! : )

Note/Warning: Due to the canvasStream bug in Mozilla, this doesn't work there.

### CODE REVIEW

File: base.js [hunk 1]
```diff
@@ -36,7 +36,12 @@ s.onload = function(){
 function storage_get_params(){
     api_base.storage.sync.get(['DogeCamConfiguration'], function(items) {
         var DogeCamConfiguration = items.DogeCamConfiguration
-
+        if(DogeCamConfiguration.draw_type==="video-playback"){
+            set_playback_element()
+        }
+        else{
```
Comment (anikumar-tivo) [inline base.js:42]: Violation of coding style 👎

File: base.js [hunk 2]
```diff
@@ -60,4 +65,55 @@ api_base.storage.onChanged.addListener(async function(changes, namespace) {
...
+    try{
+        var video_path = api_base.runtime.getURL("videos/sample.mp4")
+        console.log(video_path)
+        var playback_video = document.createElement('video')
+        playback_video.id = "playback_video"
+        playback_video.style.position = "absolute"
+        playback_video.width = 1280
+        playback_video.height = 720
+        playback_video.style.right = -1280
+        //  playback_video.style.objectFit = "fill"
```
Comment (anikumar-tivo) [inline base.js:84]: 👎

File: base.js [hunk 3]
```diff
@@ -60,4 +65,55 @@ api_base.storage.onChanged.addListener(async function(changes, namespace) {
...
+
+        playback_video.src = video_path
+        playback_video.muted = true
+        playback_video.loop = true
+        
+        playback_video.play()
+
+        document.body.appendChild(playback_div)
+        console.log("playback div added")
+        //  playback_video.play()
```
Comment (anikumar-tivo) [inline base.js:105]: 👎

File: base.js [hunk 4]
```diff
@@ -60,4 +65,55 @@ api_base.storage.onChanged.addListener(async function(changes, namespace) {
...
+    }
+    catch(error){
```
Comment (anikumar-tivo) [inline base.js:107]: violation of coding style 👎

File: base.js [hunk 5]
```diff
@@ -60,4 +65,55 @@ api_base.storage.onChanged.addListener(async function(changes, namespace) {
...
+        console.log("Static Video playback failed", error)
+    }
+    
+}
+
+function unset_playback_element(){
+    var dyn = document.getElementById("playback_div")
+    if(dyn)
+    {
```
Comment (anikumar-tivo) [inline base.js:116]: 👎

File: combined.js [hunk 6]
```diff
@@ -413,6 +413,15 @@ function drawCanvas(canvas, img, draw_type)
         //  Scale and draw to primary canvas
         scale_draw(canvas, feed)
     }
+    else if (draw_type==="video-playback"){
+        //  console.log("drawing video asset")
```
Comment (anikumar-tivo) [inline combined.js:417]: unwanted comment 👎

File: combined.js [hunk 7]
```diff
@@ -413,6 +413,15 @@ function drawCanvas(canvas, img, draw_type)
...
+        var video_asset = document.getElementById("playback_video")
+        if(!video_asset)console.log("null")
+        canvas.getContext('2d').filter="none"
+        //  console.log(video_asset.height)
```
Comment (anikumar-tivo) [inline combined.js:421]: 👎

File: combined.js [hunk 8]
```diff
@@ -413,6 +413,15 @@ function drawCanvas(canvas, img, draw_type)
...
+        scale_draw_video(canvas, video_asset)
+        //  canvas.getContext('2d').drawImage(video_asset, 0, 0)
```
Comment (anikumar-tivo) [inline combined.js:423]: 👎

File: manifest.json [hunk 9]
```diff
@@ -3,7 +3,7 @@
 
     "name": "DogeCam",
     "description": " A chrome extension to make your video calls Doge-y!",
-    "version": "0.1.3",
+    "version": "0.2.0",
```
Comment (anikumar-tivo) [inline manifest.json:6]: why 0.1.3 -> 0.2.0?

Comment (Cruzo007) [reply manifest.json:6]: A significant feature/step? We had been fixing issues on features we started with on earlier increments.

**Pull request merged**